### PR TITLE
Feature#103 resend activation api

### LINKED ISSUE #103: [Feature] Resend activation api

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #103

### CODE REVIEW

File: api/employee/config.json [hunk 1]
```diff
@@ -17,6 +17,12 @@
...
       "permission": "create employee"
     },
+    "GET /employee/:id/resend-activation": {
+      "secure": true,
+      "middleware": "resendActivation",
+      "permission": "update employee"
+    },
     "GET /employee/:employee_id/avatar": "defaultAvatar"
   }
+
```
Comment (James1x0) [inline api/employee/config.json]: ✂️

File: api/employee/index.js [hunk 2]
```diff
@@ -274,3 +274,22 @@ exports.defaultAvatar = async (ctx) => {
   ctx.type = 'image/jpeg';
   ctx.body = canvas.jpegStream();
 };
+
+exports.resendActivation = async (ctx) => {
+
```
Comment (James1x0) [inline api/employee/index.js]: ✂️

File: api/employee/index.js [hunk 3]
```diff
@@ -274,3 +274,22 @@ exports.defaultAvatar = async (ctx) => {
...
+  let id = ctx.params.id,
+      company = ctx.user.company,
+      employee = await Employee.findOne({
```
Comment (James1x0) [inline api/employee/index.js]: Break employee into it's out declaration

File: api/employee/index.js [hunk 4]
```diff
@@ -274,3 +274,22 @@ exports.defaultAvatar = async (ctx) => {
...
+        _id: id,
+        company
+      });
+
+  if (!employee){
```
Comment (James1x0) [inline api/employee/index.js]: Add space after parenth

File: api/employee/index.js [hunk 5]
```diff
@@ -274,3 +274,22 @@ exports.defaultAvatar = async (ctx) => {
...
+    ctx.status = 404;
+    ctx.body = 'employee not found';
+    return;
+  }
+  await employee.sendActivationEmail();
```
Comment (James1x0) [inline api/employee/index.js]: Add a new line before this line

File: api/employee/test.js [hunk 6]
```diff
@@ -5,34 +5,91 @@ const chai     = require('chai'),
       fs       = require('fs'),
       api      = require('../..'),
       path     = require('path'),
+      { ObjectId } = require('mongoose').Types,
+      { stub } = require('sinon'),
       File     = require('../../lib/models/file'),
       Employee = require('../../lib/models/employee'),
+      mail     = require('../../lib/mail'),
       makeUser = require('../../lib/test-support/make-user'),
       s3       = require('../../lib/load/knox')();
```
Comment (James1x0) [inline api/employee/test.js]: Realign all these declarations (select declaration lines, CMD+CNTL+/)

Comment (JavaLavaMT) [reply api/employee/test.js]: I like that shortcut!

**Pull request merged**